### ISSUE TITLE
[Bug] ChipGroup is not working in IE11 browser

### ISSUE THREAD

**Issue by karangarg45** [patternfly/patternfly-react#4148]
**Describe the issue. What is the expected and unexpected behavior?**

If I try to use chip group in an IE11 browser. A single chip without a chip group works fine though. The whole app breaks with the exception `Invalid character`
I tried digging into the code and this is what i think could be the root cause of the issue

Chip group uses `fillTemplate` function in utils
https://github.com/patternfly/patternfly-react/blob/master/packages/react-core/src/helpers/util.ts#L115

This util function creates a function on the fly which is using template literals to parse a string. My guess is babel doesn't transpile this code using a polyfill and this is why it breaks in IE11.

I also tried using this function in isolation and i could reproduce the issue

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**

This is a bug. I would say this is critical for us and not a blocker. The workaround is not to use the filter chips on IE11 by detecting the browser.

**What is your product and what release version are you targeting?**

Latest release of the patternfly and patternfly-react.

**Comment by redallen** [comment]
Hey Karan, what versions of `@patternfly/patternfly` and `@patternfly/react-core` are you using? Are you using our `patternfly-ie11.css` stylesheet?

**Comment by karangarg45** [comment]
Hi @redallen rest of the patternfly components work fine on IE11 and yes i use patternfly-ie11.css
I guess this is something related to javascript. I have also mentioned what could be the possible cause of this in my original issue description. This is the only component that breaks in IE11

**Comment by karangarg45** [comment]
The versions are 
`@patternfly/patternfly:2.71.0`
`patternfly/react-core: 3.146.0`

**Comment by redallen** [comment]
Does this issue appear on [our docs site](https://www.patternfly.org/v4/documentation/react/components/chipgroup) for `@patternfly/react-core@3.146.0`? What about the [HTML version?](https://www.patternfly.org/v4/documentation/core/components/chipgroup)

I have a feeling this is might be a CSS issue (we dropped IE11 support a few months ago). If you could share a link and/or a screenshot that'd be helpful for debugging too.

**Comment by karangarg45** [comment]
@redallen 
![image](https://user-images.githubusercontent.com/16402710/80703063-b8350f00-8aff-11ea-8703-8a4f47717057.png)

Attached screenshot for the error in the console. I would have shared the link but the website is only accessible on Red hat VPN network and on our QA env the filter chip functionality is disabled for IE and you wont be able to replicate it. 

The point i am trying to make here is that if it would have been a CSS error then why i am getting an error in the console for Javascript. The rest of the Portal Case management website works fine on IE11. If you have a portal website you can check on acess.redhat .

 I have also been trying to get your attention on the function
`
export function fillTemplate(templateString: string, templateVars: any) {
  const func = new Function(...Object.keys(templateVars), `return \`${templateString}\`;`);
  return func(...Object.values(templateVars));
}
`
Here we are creating a function dynamically using the function constructor. The return of this function would in the template string format. Since this function is created on the fly, my guess is that babel is not transpiling the code to IE11 equivalent which it does for the rest of the app.

If you want we can discuss this over a BJN call.

Also i was not aware that patternfly has dropped the IE11 support. 4-5% of our traffic comes from IE11 and according to https://access.redhat.com/help/browsers we need to support IE11.

**Comment by karangarg45** [comment]
Update

I tried removing the code which i suspected and mentioned in my above comment and i can see that my app works fine now and i am able to render the filter chip as you can see in the screenshot
![image](https://user-images.githubusercontent.com/16402710/80704888-11527200-8b03-11ea-8014-0e842a879a06.png)

SO now i am 100% sure that the filltemplate function is causing this issue. As a solution can we use a regex pattern for template filling. Something like this?
https://github.com/mikemaccana/dynamic-template/blob/master/index.js

**Comment by redallen** [comment]
Hey Karan,

You're correct! Babel transpiles to this for the ESM version:
```js
function fillTemplate(templateString, templateVars) {
  const func = new Function(...Object.keys(templateVars), `return \`${templateString}\`;`);
  return func(...Object.values(templateVars));
}
```
which throws a syntax error in IE11. I've opened #4155 to use the regex solution like you suggested. There will be a prerelease for the PR that you can use as soon as it gets merged.

In 1.5 weeks there will be one last official 3.X release that uses Babel and has your fix in it. From then on your fix will be transpiled with Typescript for our 4.X releases. I'm curious how they will fare in IE11... you can test ahead of time now at https://patternfly-react-v4.surge.sh.

**Comment by karangarg45** [comment]
Hi Allen,
Thanks for fixing this. I will test my app with PF prerelease version on Monday(Public holiday tomorrow in India)
Also thanks for the heads up on switching to typescript transpiling from babel. I will try to figure out how this will pan out for IE after the switch.

**Comment by redallen** [comment]
My pleasure @karangarg45! Give these a try when you get a chance:
```
"@patternfly/react-catalog-view-extension": "1.4.60",
"@patternfly/react-core": "3.153.15",
"@patternfly/react-inline-edit-extension": "2.17.60",
"@patternfly/react-table": "2.28.41",
"@patternfly/react-topology": "2.14.60",
"@patternfly/react-virtualized-extension": "1.4.61",
```

**Comment by tlabaj** [comment]
closed by #4155
